src/app/body/github-link/github-link.component.ts

```diff
@@ -37,16 +39,36 @@ export class GithubLinkComponent implements OnInit {
 
   submit() {
     if (this.memberOrgName) {
-      if (this.searchType == 'organisation') {
+      if (this.searchType == 'organisation' && this.linkType == 'Public') {
         this.httpService.getGithubUserRepos(this.memberOrgName).pipe(map(data => {
           const objData = data as GitOrgData[];
           return objData;
         })).subscribe(data => {
           this.objData = data;
           this.dataFetched = true;
         });
-      } 
-      else if(this.searchType == 'username') {
+      } else if (this.searchType == 'organisation') {
+        if (this.linkType=='Private'){
+          this.httpService.getGithubPrivateRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
+            this.objData = data;
+            this.dataFetched = true;
+          })
+        } else if(this.linkType=='All') {
+          this.httpService.getGithubAllRepos(this.bearerToken).pipe(map(data => {
+            const objData =data as GitOrgData[];
+            return objData;
+          })).subscribe(data => {
+            console.log(data);
```

Comment:
remove console.log and subscribe should follow the rule of next, error, complete